tradecast/chains/registry.py

```
@@ -0,0 +1,80 @@
+
+def load_chains_from_file(path: Path) -> ChainRegistry:
+    """Load chain metadata from a JSON file."""
+
+    data = json.loads(path.read_text(encoding="utf-8"))
+    if not isinstance(data, list):
+        raise TypeError("Chain configuration must be a list")
+
+    registry = ChainRegistry()
+    for entry in data:
+        if not isinstance(entry, MutableMapping):
+            raise TypeError("Each chain entry must be an object")
```

Ruff (0.13.3)

Avoid specifying long messages outside the exception class

(TRY003)